lib/community/pages/create_post_page.dart

```diff
@@ -105,12 +105,24 @@ class _CreatePostPageState extends State<CreatePostPage> {
           listener: (context, state) {
             if (state.status == ImageStatus.success) {
               _bodyTextController.text = _bodyTextController.text.replaceRange(_bodyTextController.selection.end, _bodyTextController.selection.end, "![](${state.imageUrl})");
+              setState(() => imageUploading = false);
             }
             if (state.status == ImageStatus.successPostImage) {
               _urlTextController.text = state.imageUrl;
+              setState(() => postImageUploading = false);
```

Comment:
This is probably a question that @hjiangsu can answer better, but are we mixing styles here by calling `setState` from a `BlocListener`? I believe the builder will be called again when the state changes, so can we check `state.status` in the `IconButton` rather than using a separate property and rebuilding again?

Reply:
Hmm, if we are only using BlocListener, I don't know if we have access to the state from the child. However, if we were using a BlocBuilder or BlocConsumer, then I would agree with this.

I think its okay to set states in here given that we're not using BlocBuilder/BlocConsumer